sds_data_manager/stacks/ialirt_database_stack.py

```diff
@@ -0,0 +1,54 @@
+"""Stack for creating the IALIRT database."""
+
+from aws_cdk import (
+    RemovalPolicy,
+    Stack,
+)
+from aws_cdk import (
+    aws_dynamodb as ddb,
+)
+from constructs import Construct
+
+
+class IAlirtDatabaseStack(Stack):
+    """Stack for creating the IALIRT database."""
+
+    def __init__(self, scope: Construct, construct_id: str, **kwargs) -> None:
+        """Initialize the IALIRT database stack.
+
+        Parameters
+        ----------
+        scope : Construct
+            The App object in which to create this Stack
+        construct_id : str
+            The ID (name) of the stack
+        **kwargs
+            Additional keyword arguments.
+
+        Note:
+        DynamoDB's table defines the key schema and any secondary indexes.
+        The non-key attributes are included items are added to the table.
+        """
+        super().__init__(scope, construct_id, **kwargs)
+
+        self.packet_data_table = ddb.Table(
+            self,
+            "PacketDataTable",
+            table_name="imap-packetdata-table",
+            # Change to RemovalPolicy.RETAIN to keep the table after stack deletion.
+            # TODO: change to RETAIN in production.
+            removal_policy=RemovalPolicy.DESTROY,
+            # Restore data to any point in time within the last 35 days.
+            # TODO: change to True in production.
+            point_in_time_recovery=False,
+            # Partition key (PK) = spacecraft time ugps.
+            partition_key=ddb.Attribute(
+                name="sct_vtcw_reset#sct_vtcw",
```

Comment:
I think this needs to match your Lambda insert function above.

```suggestion
                name="reset_number#met",
```

I also think that including "reset" in the key might be hard to include in your queries and what happens when a reset does happen, will your processing code have to do a scan over all items to make sure we are still on reset 0, and where do you get that information from in the CCSDS packet? I think all you'll have access to is the SCLK / MET.

Reply:
From conversations with Steve Monk a reset would be a big deal. For now I plan on assigning a value in the lambda to a reset_number = 0. If a reset does occur we will be notified and I would have to manually go in, change the reset_number in the lambda, and redeploy.